Frameworks/Foundation/NSOperation.mm

```diff
@@ -84,11 +84,32 @@ - (void)_checkReady {
     }
 }
 
+- (void)_completionBlockThread {
+    decltype(_completionBlock) localCompletion;
+
+    { // _completionBlockLock scope
+        std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+        localCompletion = std::move(_completionBlock);
+    } // end _completionBlockLock scope
+
+    if (localCompletion) {
+        localCompletion();
+    }
+}
+
+- (void)_spawnCompletionBlockThread {
+    // Occurs under _finishLock.
+    [NSThread detachNewThreadSelector:@selector(_completionBlockThread) toTarget:self withObject:nil];
```

Comment:
>detachNewThreadSelector [](start = 14, length = 23)

We will spin off a new thread for every operation that completes, which is not good.  At the very least, we should use a thread pool, ideally we should further complicate the code and call it in the NSOperationQueue thread. :)